kitsune/sumo/utils.py

```diff
@@ -128,6 +128,10 @@ def get_next_url(request):
     else:
         url = request.META.get("HTTP_REFERER")
 
+    # Remove any newline or carriage-return characters.
+    if url:
+        url = url.replace("\n", "").replace("\r", "")
```

Comment:
The `\t` character also needs to be removed, according to https://url.spec.whatwg.org/#url-parsing. However, I'd tend to simply reject URLs containing these characters instead of trying to sanitize them.